### PR TITLE
feat: add support for unegotiated incoming groups

### PR DESCRIPTION
**Pull request by murillo128**
This PR adds support for detecting unsignaled sources that provides relevant `mid` and `rid` values.

The `mid` is used for selecting the track to associate the id, and the `rid` for the source id. RTX ssrc is associated by `rid` or `rrid` afterwards.

Also I had to change some raw pointers by shared pointers.

### CODE REVIEW

File: include/DTLSICETransport.h [hunk 1]
```diff
@@ -179,10 +180,10 @@ class DTLSICETransport :
 	WORD		feedbackCycles			= 0;
 
 	//TODO: change by shared pointers
```
Comment (eck1001) [inline include/DTLSICETransport.h:182]: can this todo be removed now?

File: include/rtp/RTPOutgoingSourceGroup.h [hunk 2]
```diff
@@ -63,6 +63,7 @@ struct RTPOutgoingSourceGroup
 	}
 	
 public:	
+	std::string rid;
```
Comment (eck1001) [inline include/rtp/RTPOutgoingSourceGroup.h:66]: question: rid is "repairedStreamId" correct? What caused an id to need repairing?

File: include/rtp/RTPOutgoingSourceGroup.h [hunk 3]
```diff
@@ -63,6 +63,7 @@ struct RTPOutgoingSourceGroup
...
 	std::string mid;
```
Comment (eck1001) [inline include/rtp/RTPOutgoingSourceGroup.h:67]: question: `mid` is mediaStreamId?

File: include/rtp/RTPOutgoingSourceGroup.h (hunk 2 above)
Comment (murillo128) [reply include/rtp/RTPOutgoingSourceGroup.h:66]: that's the name of the rtp header extension (https://www.rfc-editor.org/rfc/rfc8852.html#name-rtcp-repairedrtpstreamid-sd) for RTX.  rid is the rtp-stream-id and the rrid is the repaired-rtp-stream-id.

When receiving a RTX packate with an rrid value, you can find the media stream by checking  rtx rrid = media rid.

File: include/rtp/RTPOutgoingSourceGroup.h (hunk 3 above)
Comment (murillo128) [reply include/rtp/RTPOutgoingSourceGroup.h:67]: yes

File: include/DTLSICETransport.h (hunk 1 above)
Comment (murillo128) [reply include/DTLSICETransport.h:182]: yes!

**Pull request merged**